File: substrate_spatial_browser/android/app/src/main/kotlin/com/example/substrate_spatial_browser/MultisetVpsPlugin.kt
package com.example.substrate_spatial_browser

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

class MultisetVpsPlugin : FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(
            binding.binaryMessenger,
            "com.substrate.multiset/vps"
        )
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            "startVPSSession" -> {
                // Phase 5: Stub returns true
                // Phase 6: Initialize MultiSet Android SDK with call.argument("apiKey")
                result.success(true)
            }
            "stopVPSSession" -> {
                // Phase 6: Terminate MultiSet SDK session
                result.success(null)
            }
            "getCurrentPose" -> {
                // Phase 5: Return mock pose data for development
                // Phase 6: Return real pose from MultiSet SDK
                result.success(mapOf(
                    "lat" to 29.4241,
                    "lng" to -98.4936,
                    "alt" to 195.2,
                    "accuracy_m" to 0.05,
                    "yaw" to 0.0,
                    "pitch" to 0.0,
                    "roll" to 0.0
                ))
            }
            "createAnchor" -> {
                // Phase 5: Return stub anchor ID
                // Phase 6: Create real anchor via MultiSet Android SDK
                result.success("multiset-anchor-stub-${System.currentTimeMillis()}")
            }
            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}

File: substrate_spatial_browser/ios/Runner/MultisetVpsPlugin.swift
import Flutter
import UIKit

public class MultisetVpsPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(
            name: "com.substrate.multiset/vps",
            binaryMessenger: registrar.messenger()
        )
        let instance = MultisetVpsPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "startVPSSession":
            // Phase 5: Stub returns true
            // Phase 6: Initialize MultiSet iOS SDK
            result(true)
        case "stopVPSSession":
            // Phase 6: Terminate MultiSet iOS SDK session
            result(nil)
        case "getCurrentPose":
            // Phase 5: Return mock pose for development
            // Phase 6: Return real pose from MultiSet iOS SDK
            result([
                "lat": 29.4241,
                "lng": -98.4936,
                "alt": 195.2,
                "accuracy_m": 0.05,
                "yaw": 0.0,
                "pitch": 0.0,
                "roll": 0.0
            ])
        case "createAnchor":
            // Phase 5: Return stub anchor ID
            let stubId = "multiset-anchor-stub-\(Int(Date().timeIntervalSince1970 * 1000))"
            result(stubId)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}
